### PR TITLE
Add FXIOS-8472 [v124] Create SwiftUI components for header and footer to share between login and address autofill

### PR DESCRIPTION
**Pull request by razvanlitianu**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8472)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18794)

## :bulb: Description

Addresses
Screenshot 2024-02-21 at 11 51 19
Passwords
Screenshot 2024-02-21 at 11 50 54
Passwords footer
Screenshot 2024-02-21 at 10 54 48

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Review by jnrahme — APPROVED** [review]

**Review by nbhasin2 — APPROVED** [review]
This looks fine, and yes it can pose some accessibility concern for which we can add another PR during integration 

@dataports feel free to look around :)

**Comment by nbhasin2** [comment]
Going to merge this as its a blocker for @jnrahme 
@dataports feel free to leave comments and we will address them in a following PR

File: firefox-ios/Client/Frontend/Autofill/AutoFillFooterView.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import SwiftUI
+import Common
+
+struct AutoFillFooterView: View {
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Autofill/AutoFillFooterView.swift:8]: I'm sorry for the nit, `AutoFill` should be `Autofill` just to be consistent with the rest of the codebase and easier to search.

File: firefox-ios/Client/Frontend/Autofill/AutoFillHeaderView.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import SwiftUI
+import Common
+
+struct AutoFillHeaderView: View {
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Autofill/AutoFillHeaderView.swift:8]: same nit as above

File: firefox-ios/Client/Frontend/Autofill/AutoFillHeaderView.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    @State private var textPrimary: Color = .clear
+    @State private var textSecondary: Color = .clear
+
+    @Environment(\.themeType)
+    var theme
+
+    var title: String
+    var subtitle: String?
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Autofill/AutoFillHeaderView.swift:26]: both vars can be private

File: firefox-ios/Client/Frontend/Autofill/AutoFillHeaderView.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                Image(uiImage: UIImage(imageLiteralResourceName: ImageIdentifiers.homeHeaderLogoBall))
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: UX.logoSize, height: UX.logoSize)
+                VStack(alignment: .leading) {
+                    Text(title)
+                        .font(.body)
+                        .fontWeight(.bold)
+                        .foregroundColor(textPrimary)
+                    subtitle.map {
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Autofill/AutoFillHeaderView.swift:45]: Interesting way to not show a view on a nil value 👀

**Comment by dataports** [comment]
Looks great ⭐ , sorry for the late review, just left some very small bits of feedback.

**Pull request merged**